arangod/VocBase/vocbase.cpp

```diff
@@ -81,6 +97,108 @@
 using namespace arangodb;
 using namespace arangodb::basics;
 
+struct arangodb::VocBaseLogManager {
+  explicit VocBaseLogManager(application_features::ApplicationServer& server, DatabaseID database)
+      : _server(server),
+        _logContext(LoggerContext{Logger::REPLICATION2}.with<logContextKeyDatabaseName>(
+            std::move(database))) {}
+
+  [[nodiscard]] auto getReplicatedLogById(replication2::LogId id) const
+      -> std::shared_ptr<replication2::replicated_log::ReplicatedLog const> {
+    if (auto iter = _logs.find(id); iter != _logs.end()) {
+      return iter->second;
+    }
+    // THROW_ARANGO_EXCEPTION_PARAMS(TRI_ERROR_REPLICATION_REPLICATED_LOG_NOT_FOUND, id.id());
+    THROW_ARANGO_EXCEPTION_FORMAT(TRI_ERROR_REPLICATION_REPLICATED_LOG_NOT_FOUND,
+                                  "replicated log %" PRIu64 " not found", id.id());
+  }
+
+  [[nodiscard]] auto getReplicatedLogById(replication2::LogId id)
+      -> std::shared_ptr<replication2::replicated_log::ReplicatedLog> {
+    if (auto iter = _logs.find(id); iter != _logs.end()) {
```

Comment:
I think we need to protect `_logs` from concurrent modifications by acquiring the `_mutex`.

Reply:
Fixed